### PR TITLE
fix(datastore): Ignore field mismatch errors

### PR DESCRIPTION
**Pull request by bhshkh**
ErrFieldMismatch is returned when a field is to be loaded into a different type than the one it was stored from, or when a field is missing or unexported in the destination struct.

This PR allows user to suppress such errors. Some scenarios in which this would be useful are listed in https://github.com/googleapis/google-cloud-go/issues/913

Points to note
1. Suppressing the error does not mean that Datastore will not fetch missing field. It just means that the field was still fetched from Datastore service but not loaded in user's struct.
2. When writing the data back to Datastore, it will be overwritten. E.g.
If entity written to Datastore is `{X: 1, Y:2}` and it is loaded into `type NewStruct struct{X int}`, then `{X:1}` is returned to the user. Now, if user modifies this `X = 5` and writes it back to Datastore, there will be no Y field left for this entity in Datastore.

Fixes: https://github.com/googleapis/google-cloud-go/issues/913

### CODE REVIEW

File: datastore/datastore.go [hunk 1]
```diff
@@ -357,6 +357,54 @@ func checkMultiArg(v reflect.Value) (m multiArgType, elemType reflect.Type) {
 	return multiArgTypeInvalid, nil
 }
 
+// IgnoreFieldMismatch allows ignoring ErrFieldMismatch error while
+// reading or querying data.
+// WARNING: Ignoring ErrFieldMismatch can cause data loss
+func (c *Client) IgnoreFieldMismatch() {
```
Comment (codyoss) [inline datastore/datastore.go:363]: Consider expressing this as a client specific ClientIOption. Here is an example from storage: https://github.com/googleapis/google-cloud-go/blob/532a37c98680855fa38342bb7680062226594a0e/storage/option.go#L63-L75

Comment (bhshkh) [reply datastore/datastore.go]: Updated the PR

**Comment by codyoss** [comment]
@bhshkh do we still want this PR?

File: datastore/option.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+// A datastoreClientOption is an option for a Google Datastore client.
+type datastoreClientOption interface {
+	option.ClientOption
+	applyDatastoreOpt(*datastoreConfig)
+}
+
+// WithIgnoreFieldMismatch allows ignoring ErrFieldMismatch error while
+// reading or querying data.
+// WARNING: Ignoring ErrFieldMismatch can cause data loss
+func WithIgnoreFieldMismatch() option.ClientOption {
```
[the remark this answers is not in the document]
Comment (bhshkh) [reply datastore/option.go:54]: General naming convention for options is WIthXXXX
Some examples
https://github.com/googleapis/google-cloud-go/blob/main/vertexai/genai/option.go#L22-L26

https://github.com/googleapis/google-cloud-go/blob/main/storage/transfermanager/option.go#L27-L62

**Review by gkevinzheng — APPROVED** [review]

**Pull request merged**